### ISSUE TITLE
Framework for controlling output and order of output of ILL Module tables/lists

### ISSUE THREAD

**Issue by a-sassmannshausen** [PTFS-Europe/koha#69]
Chatting to Martin about Koha's XSLT theming made me think that it could be very feasible to add a further configuration layer (perhaps as yaml) that specifies what fields should be shown on what pages in the ILL module.

We would operate on 2 levels:
- spec.yaml specifies what data should be passed to the front-end, depending on "Summary / Full" view.
- display.yaml specifies what fields should be displayed on which pages, and in what order they should be presented.

**Comment by a-sassmannshausen** [comment]
The framework should allow us to specifiy:
- title (human readable)
- order (perhaps simple numbering for each entry, 1 first, 10 later)
- hide/show
- width (percentage?)
- color (inline css?)

**Comment by janetc** [comment]
Is part of this enabling different fields to show dependent on which material type the end user selects? E.g. select Article and you are only presented with fields relevant to Article requests (particulary for Manual entry form)?

**Comment by a-sassmannshausen** [comment]
Indeed, part of this framework should be an identifier that would allow us to do this.  (either by getting the user to choose the type before getting to the search form proper (dislike this idea), or by tagging the form in such a way, and making type the first field in the form, such that when the type is selected, some jquery could intelligently remove non-relevant fields.

This would be particularly interesting for the manual create form.

The table displays of existing requests and search results would also benefit indirectly, as it would provide a robust way to split the page into multiple tables, one for each item type.
